extensions/wikia/PartnerFeed/PartnerFeed.body.php

```diff
@@ -81,100 +77,105 @@ function execute() {
+	 */
+	private function FeedRecentBlogComments ( $format ) {
+		$wg = F::app()->wg;
+
+		if ( empty( $wg->EnableBlogArticles ) ) {
 			$this->showMenu();
 		} else {
-
 			// local settings
 			$maxNumberOfBlogComments = 10;
 			$userAvatarSize = 48;
 
-			$sBlogPost = $wgRequest->getText ( "blogpost", false );
-			$oTitle = Title::newFromText( $sBlogPost , 500);
-			if ( $oTitle->getArticleID() > 0 ){
+			$sBlogPost = $wg->Request->getText ( "blogpost", false );
+			$oTitle = Title::newFromText( $sBlogPost , 500 );
+			if ( $oTitle && $oTitle->getArticleID() > 0 ) {
 
-				$articleCommentList = ArticleCommentList::newFromTitle($oTitle);
+				$articleCommentList = ArticleCommentList::newFromTitle( $oTitle );
 				$articleCommentList->newFromTitle( $oTitle );
 				$aCommentPages = $articleCommentList->getCommentPages();
 
-				$counter = $maxNumberOfBlogComments;
-				$feedArray = array();
-				foreach ($aCommentPages as $commentPage){
+				$feedArray = [];
+				foreach ( $aCommentPages as $commentPage ) {
 
-					if ( ( $maxNumberOfBlogComments-- ) == 0){
+					if ( ( $maxNumberOfBlogComments-- ) == 0 ) {
 						break;
 					}
-					//make sure all data is loaded
-					if ( $commentPage[ 'level1' ]->load() ) {
-						$tmpArticleComment = $commentPage[ 'level1' ]->getData();
 
-						$feedArray[ ] = array(
+					/** @var ArticleComment $levelOne */
+					$levelOne = $commentPage['level1'];
+
+					// make sure all data is loaded
+					if ( $levelOne->load() ) {
+						$tmpArticleComment = $levelOne->getData();
+
+						/** @var Revision $revision */
+						$firstRevision = $levelOne->mFirstRevision;
+						if ( !$firstRevision instanceof Revision ) {
+							$this->logError( 'Comment has no first revision', $tmpArticleComment );
+							break;
+						}
+
+						/** @var User $commentAuthor */
+						$commentAuthor = $tmpArticleComment[ 'author' ];
+						$feedArray[] = [
 							'title' => '',
 							'description' => $tmpArticleComment[ 'text' ],
 							'url' => $oTitle->getFullURL(),
-							'date' => $commentPage[ 'level1' ]->mFirstRevision->getTimestamp(),
-							'author' => $tmpArticleComment[ 'author' ]->getName(),
-							'otherTags' => array(
-								'image' => AvatarService::getAvatarUrl( $commentPage[ 'level1' ]->mUser->getName(), $userAvatarSize )
-							)
-						);
+							'date' => $firstRevision->getTimestamp(),
+							'author' => $commentAuthor->getName(),
+							'otherTags' => [
+								'image' => AvatarService::getAvatarUrl( $levelOne->mUser->getName(), $userAvatarSize )
+							]
+						];
 					} else {
-						\Wikia\Logger\WikiaLogger::instance()->error('Comment page loading failed', $commentPage);
+						$this->logError( 'Comment page loading failed', $commentPage );
 					}
 				}
 
-				$this->showFeed( $format , wfMsg( 'feed-title-blogcomments', $oTitle->getFullText() ),  $feedArray);
-
+				$this->showFeed(
+					$format,
+					wfMessage( 'feed-title-blogcomments', $oTitle->getFullText() )->text(),
```

Comment:
Yah, RSS and Atom feeds use CDATA blocks to escape content which means we don't need to escape these for transport.  The lines that include user content (like this one) are ok because the content is all titles which are already sanitized for special characters.